Flutter_Weather/lib/models/weather_extension.dart

```diff
@@ -0,0 +1,94 @@
+import 'dart:math';
+
+import 'package:flutter_weather/models/index.dart';
+import 'package:flutter_weather/resources/assets.dart';
+import 'package:flutter_weather/services/date_formatter.dart';
+import 'package:intl/intl.dart';
+
+extension TimeHandlerExtension on Weather{
+  get initDate => yearMonthDayHour(init);
+
+  DateTime getTrueDate(int timePoint) => this.initDate.add(Duration(hours: timePoint));
+
+  List<DateTime> getTrueDates(List<int> weekDayIndex)
+  => weekDayIndex.map((index) => this.getTrueDate(dataseries[index].timepoint)).toList();
+
+  DateTime yearMonthDayHour(String initDate){
+    final year = initDate.substring(0, 4);
+    final month = initDate.substring(4, 6);
+    final day = initDate.substring(6, 8);
+    final hour = initDate.substring(8, initDate.length);
+    return DateFormat('yyyy MM dd hh').parse("$year $month $day $hour", true);
+  }
+}
+
+extension DayWeatherExtension on Weather{
+  DayWeather get dayData{
```

Comment:
Remove this.
Create List<DayWeather> allDays; and List<DayWeather> availableDays;